### ISSUE TITLE
Native mode of DNS encryption and SDNS stamps

### ISSUE THREAD

**Issue by ammnt** [AdguardTeam/AdguardForiOS#2103]
### Prerequisites

Please answer the following questions for yourself before submitting an issue. **YOU MAY DELETE THE PREREQUISITES SECTION.**

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed

### Issue Details

- **AdGuard version:**
  - AdGuard for iOS v4.4.5 (990)
- **Device model and storage size:**
  - iPhone XR
- **Operating system and version:**
  - 16.2

### Expected Behavior

DoH encryption works without initial resolving as bootstrap IP is predefined.

### Actual Behavior

Resolving with SDNS stamps doesn't work and queries is leaked, except .arpa domains.

### Additional Information

I expected that bootstrap IP server from SDNS stamp will be work just like the lines "ServerAddresses" in .mobileconfig. For example:
```
<key>DNSProtocol</key>
<string>HTTPS</string>
<key>ServerAddresses</key>
    <array>
        <string>8.8.4.4</string>
        <string>8.8.8.8</string>
    </array>
<key>ServerURL</key>
<string>https://dns.google/dns-query</string>
```

**Comment by Versty** [comment]
@ammnt Are you talking about Native implementation?

**Comment by ammnt** [comment]
> @ammnt Are you talking about Native implementation?

Yes. I just to be sure that it's make sense to add SDNS stamps with the predefined bootstrap IP address if i use native mode for DoH.

**Comment by Versty** [comment]
@ammnt This behaviour will be improved in the future versions.
